neptune/new/internal/backends/operations_preprocessor.py

```diff
@@ -65,7 +66,7 @@ def process(self, operations: List[Operation]):
 
     def get_operations(self) -> List[Operation]:
         result = []
-        for _, acc in sorted(self._accumulators.items()):
+        for _, acc in self._accumulators.items():
```

Comment:
We dropped sort, because we have to make sure that every batch is limited by last `copy` operation, right?

Why did we sorted in the first place?

Reply:
1. yes, exactly
2. that's a great question. @aniezurawski maybe you know?

Reply:
I do not remember. It is sorted by path so it should not metter. Maybe it was easier to test it this way?

Anyway, are you guys sure it is good idea to change anything in operation_preprocessor? Shouldn't be part-batches preprocessed independently?

Reply:
I think you could just split a batch by copy operations and reuse existing code for every part-batch. Much simpler and elegant.

Reply:
Moreover we probably should ack after every part-batch, so maybe single call to execute_operations should just return number operations actually performed and do a single part-batch at the time?

Reply:
That would change the idea that in processing a queue batch we do upload (and artifact) ops first -- we would do upload "subbatch", then other "subbatch", and upload again (there's a special comment saying "Upload operations should be done first since they are idempotent").

I'm not if we actually need that -- other than that, I'm perfectly OK with what you suggest.

Reply:
> That would change the idea that in processing a queue batch we do upload (and artifact) ops first

No, it would not. It would still apply but to potentially smaller batches. Just split first, and do upload-related logic later for every part-batch. ;)

Reply:
> Moreover we probably should ack after every part-batch, so maybe single call to execute_operations should just return number operations actually performed and do a single part-batch at the time?

Right, that sounds good (although the "much simpler" may be lost on me ;))

Reply:
done